base/file.jl

```diff
@@ -484,6 +484,7 @@ function temp_cleanup_purge(; force::Bool=false)
             if (force || asap) && ispath(path)
                 need_gc && GC.gc(true)
                 need_gc = false
+                chmod(path, 0o777; recursive=true)
```

Comment:
I've been changing to `0o700` instead so that we don't accidentally expose these files to being read by an eavesdropper. I guess the issue with that is that the user might not own the files but have write permissions and take away their own write permissions by doing this. But I think it the user doesn't own the files, that's a corner case where we can just let this fail.

Reply:
The fully correct way to do this would be to add an option to `rm` that lets you try harder and changes the permissions precisely as necessary to allow deletion. So that would entail checking who owns the file and and picking the permissions based on that.

Reply:
What about just setting all directories to `0o770`?  I was being a little lazy by using `chmod(; recursive=true)` here, we really only care about the directories.

Reply:
Making something group-readable is not really safer than world. What about `0o722`? That makes it writeable by anyone, which is normally a security hole but in this case since we're about to delete it, it should be safe.

Reply:
Okay the actual change I made was to set it to `filemode(path) | 0o220`.  That seems the best option.

Reply:
I don't think there's a meaningful distinction between group and world on most systems: the default group is usually the group that everyone belongs to (except for root). So what about `0o222`? Giving someone else write permissions is acceptable since we're about to delete the file and so writing to it can't cause trouble.